packages/editor/src/components/block-switcher/index.js

```diff
@@ -36,19 +36,27 @@ export class BlockSwitcher extends Component {
 	}
 
 	render() {
-		const { blocks, onTransform, isLocked } = this.props;
+		const { blocks, onTransform, inserterItems } = this.props;
 		const { hoveredClassName } = this.state;
 
 		if ( ! blocks || ! blocks.length ) {
 			return null;
 		}
 
-		const allowedBlocks = getPossibleBlockTransformations( blocks );
+		const itemsByName = mapKeys( inserterItems, ( { name } ) => name );
+		const possibleBlockTransformations = sortBy(
+			filter(
+				getPossibleBlockTransformations( blocks ),
+				( block ) => !! itemsByName[ block.name ]
+			),
+			( block ) => -itemsByName[ block.name ].frecency,
```

Comment:
We can make it a little more explicit that this is a reverse sort by using [`orderBy`](https://lodash.com/docs/#orderBy).

```js
const possibleBlockTransformations = orderBy(
	...,
	( block ) => itemsByName[ block.name ].frecency,
	'desc'
);
```